sdk/simulator-service/src/implementation/simulator-engine/simulator.ts

```diff
@@ -23,23 +25,49 @@ export class Simulator<
   }
 
   static create<S extends SimulationStrategy>(schema: S) {
-    return new Simulator(schema)
+    return new Simulator(schema, schema)
```

Comment:
is this a mistake?

Reply:
No it's not, I realized that I need to keep the original schema for processing. It works in such way that when adding new step it removes first step from schema for typing purposes I wanted Simulator to be a generic that with Schema array being reduced each time you add next step.

I guess I could have done it differently, and just keep the schema and index of current processing step.

Reply:
Do you mind adding a comment to that? It's going to raise some questions in the future I'm afraid. We can talk about it when you are back. Thanks!

Reply:
I can refactor it one I'm back